src/modules/universal-discovery/components/content-table/content.table.component.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ContentTablePaginationComponent from './content.table.pagination.component';
+import ContentTableItemComponent from './content.table.item.component';
+import ContentTableHeaderComponent from './content.table.header.component';
+
+import './css/content.table.component.css';
+
+export default class ContentTableComponent extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            items: props.items,
+            perPage: props.perPage,
+            activePage: 0,
+            pages: this.splitToPages(props.items, props.perPage)
+        };
+
+        this.setActivePage = this.setActivePage.bind(this);
+        this.renderItem = this.renderItem.bind(this);
+    }
+
+    componentWillReceiveProps({ items, perPage }) {
+        this.setState(state => ({
+            ...state,
+            items,
+            pages: this.splitToPages(items, perPage)
+        }));
+    }
+
+    /**
+     * Splits items into pages
+     *
+     * @method splitToPages
+     * @param {Array} items
+     * @param {Number} perPage
+     * @returns {Array}
+     * @memberof ContentTableComponent
+     */
+    splitToPages(items, perPage) {
+        return items.reduce((pages, item, index) => {
+            const pageIndex = Math.floor(index / perPage);
+
+            if (!pages[pageIndex]) {
+                pages[pageIndex] = [];
+            }
+
+            pages[pageIndex].push(item);
+
+            return pages;
+        }, []);
+    }
+
+    /**
+     * Sets active page index state
+     *
+     * @method setActivePage
+     * @param {number} activePage
+     * @memberof ContentTableComponent
+     */
+    setActivePage(activePage) {
+        this.setState(state => ({
+            ...state,
+            activePage
+        }));
+    }
+
+    /**
+     * Renders single item
+     *
+     * @method renderItem
+     * @param {Object} item
+     * @returns {Element}
+     * @memberof ContentTableComponent
+     */
+    renderItem(item) {
+        const location = item.Location;
+
+        const { contentTypesMap, onItemSelect, onItemClick, labels } = this.props;
+
+        return <ContentTableItemComponent
+            key={location.id}
+            data={location}
+            contentTypesMap={contentTypesMap}
+            onPreview={onItemSelect}
+            onItemClick={onItemClick}
+            labels={labels.item} />;
+    }
+
+    /**
+     * Renders no items message 
+     *
+     * @method renderNoItemsMessage
+     * @returns {Element}
+     * @memberof ContentTableComponent
+     */
+    renderNoItemsMessage() {
+        const { items } = this.state;
+        const { noItems } = this.props.labels;
+        const showMessage = items.length === 0 && !!noItems;
```

Comment:
I think it will be ok with `!items.length && !!noItems`

Reply:
And do we need to check if `noItems` label is set? I mean this one is required.

Reply:
I removed `isRequired` flag form `propTypes`.